lib/test.js

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2012-2018 Digital Bazaar, Inc. All rights reserved.
+ */
+const bedrock = require('bedrock');
+const Server = require('karma').Server;
+
+// load config defaults
+require('./config');
+
+bedrock.events.on('bedrock.tests.run', (state) => {
+  if(bedrock.test.shouldRunFramework('karma')) {
+    return run(state);
+  }
+});
+
+function run(state) {
+  return new Promise(resolve => {
+    const command = bedrock.config.cli.command;
+    if(!command.karmaSuite &&
+      Object.keys(bedrock.config.karma.suites).length === 0) {
+      // no tests to run
+      console.log('No karma tests to run.');
+      // print eol
+      console.log();
+      resolve();
+    }
+
+    const karmaConfig = generateKarmaConfig();
+
+    const server = new Server(karmaConfig);
+
+    server.on('run_complete', (browsers, results) => {
+      if(results.failed > 0) {
+        state.pass = false;
+      }
+      resolve();
+    });
+
+    console.log('Running tests via Karma...');
+    server.start();
+  });
+}
+
+function generateKarmaConfig() {
+  const {karma: {config: karmaConfig}} = bedrock.config;
+  const {karma: {defaults: {DEFAULT_PREPROCESSORS}}} = bedrock.config;
```

Comment:
This could probably be done with a single declaration. Not sure if that's a good idea or not -- haven't settled on a good pattern.